### PR TITLE
Add integration with GraphQL

### PR DESCRIPTION
**Pull request by c-w**
This pull request adds a connection and data-source for GraphQL.

To test the integration, place the following content into a file called `graphql.private.js` in the `server/dashboards` folder:

```js
return {
  id: "graphql",
  name: "GraphQL",
  icon: "dashboard",
  url: "graphql",
  description: "A basic sample to see how a GraphQL service is connected to graphs",
  preview: "/images/bot-framework-preview.png",
  config: {
    connections: {
      graphql: { serviceUrl: "https://pokeapi-graphiql.herokuapp.com" }
    },
    layout: {
      isDraggable: true,
      isResizable: true,
      rowHeight: 30,
      verticalCompact: false,
      cols: { lg: 12, md: 10, sm: 6, xs: 4, xxs: 2 },
      breakpoints: { lg: 1200, md: 996, sm: 768, xs: 480, xxs: 0 }
    }
  },
  dataSources: [
    {
      id: "pokeIds",
      type: "Constant",
      params: { values: [12, 13, 14], selectedValue: 12 },
      calculated: (state, dependencies) => {
        return { variables: { pokeId: state.selectedValue } };
      }
    },
    {
      id: "graphql",
      type: "GraphQL",
      dependencies: { variables: "pokeIds:variables" },
      params: { 
        query: "query($pokeId:Int!){pokemon(number:$pokeId){name}}"
      },
      calculated: (state, dependencies) => {
        return { pokemons: [state.data.pokemon] };
      }
    }
  ],
  filters: [
    {
      type: "TextFilter",
      title: "PokeId",
      dependencies: { selectedValue: "pokeIds", values: "pokeIds:values" },
      actions: { onChange: "pokeIds:updateSelectedValue" },
      first: true
    },
  ],
  elements: [
    {
      id: "names",
      type: "Table",
      size: { w: 4, h: 8 },
      dependencies: {
        values: "graphql:pokemons"
      },
      props: {
        cols: [
          { header: "Name", field: "name" }
        ]
      },
    }
  ],
  dialogs: []
}
```

Open the thus-defined [dashboard](http://localhost:3000/dashboard/graphql) to fetch the name for a Pokemon from the [GraphQL PokeAPI](https://pokeapi-graphiql.herokuapp.com) and display it in an Ibex table:

image

### CODE REVIEW

**Comment by morsh** [comment]
@c-w not sure if this should be approved prior event hub. Plus, add this sample (minus the connection string) to the dashboard samples once you added event hub integration and add html instructions how to add the connection string (under connection => editor)

File: src/data-sources/plugins/GraphQL.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+    // Validate connection
+    const connection = this.getConnection();
+    const { serviceUrl } = connection;
+    if (!connection || !serviceUrl) {
+      return dispatch => dispatch();
+    }
+
+    const params = this.getParams();
+    const query = params.query;
```
Comment (morsh) [inline src/data-sources/plugins/GraphQL.ts]: const params = this.getParams() || {};
const { query } = params || {};

File: src/data-sources/plugins/GraphQL.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        }
+
+        return dispatch(json);
+      });
+    };
+  }
+
+  updateSelectedValues(dependencies: IDictionary, selectedValues: any) {
+    if (Array.isArray(selectedValues)) {
+      return Object.assign(dependencies, { 'selectedValues': selectedValues });
```
Comment (morsh) [inline src/data-sources/plugins/GraphQL.ts]: I find that using lodash is more readable, especially if you have it in your project. But that is just a recommendation, not a must. Plus, you can use a simpler form of assign to property:

Object.assign(dependencies, { selectedValues });

Comment (c-w) [inline src/data-sources/plugins/GraphQL.ts]: This is copied from other `updateSelectedValues` implementations (e.g. [Sample.ts](https://github.com/CatalystCode/ibex-dashboard/blob/d89db0cbf2184d9efaa82c48f0a13904a3ef251b/src/data-sources/plugins/Sample.ts#L41-L47), [Azure.ts](https://github.com/CatalystCode/ibex-dashboard/blob/d89db0cbf2184d9efaa82c48f0a13904a3ef251b/src/data-sources/plugins/Azure.ts#L97-L103)) so I'd say that we should clean this up in bulk in the future.

File: src/data-sources/plugins/GraphQL.ts (hunk 1 above)
Comment (c-w) [inline src/data-sources/plugins/GraphQL.ts]: The suggested code doesn't compile, but I managed to add default values in e21d714.

**Pull request merged**